Python/sysmodule.c

```diff
@@ -1941,6 +1941,37 @@ sys__clear_type_cache_impl(PyObject *module)
     Py_RETURN_NONE;
 }
 
+#ifdef Py_DEBUG
+/*[clinic input]
+sys._get_type_version_tag
+
+    type: object
+    /
+
+For internal use only: get the `tp_version_tag` of *type*.
+[clinic start generated code]*/
+
+static PyObject *
+sys__get_type_version_tag(PyObject *module, PyObject *type)
+/*[clinic end generated code: output=bdcfab8b295e0a76 input=6016e5a02b3c3675]*/
+{
+    PyThreadState *tstate = _PyThreadState_GET();
+    _Py_EnsureTstateNotNULL(tstate);
```

Comment:
You can use PyThreadState_Get() which does the assert for you.

Reply:
I just did that in https://github.com/python/cpython/pull/27774/commits/8a01d9c318cf11c08c71bbfd4562c3c0f77bfea5 a few seconds before you wrote this message :).